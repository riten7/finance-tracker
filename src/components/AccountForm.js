import React from 'react';
import { Form } from 'semantic-ui-react';
import { accountTypes, getRandomId } from '../utility/constant';

const AccountForm = (props) => {

  const handleSubmit = () => {
    props.handleFormSubmit({
      ...accountDetail, 
      id: getRandomId(),
    });
    resetForm();
  }

  const [accountDetail, setAccountDetail] = React.useState({
    name: '',
    type: accountTypes[0].value,
    balance: '',
  });
  const handleChange = (_, { name, value }) => {
    setAccountDetail({
      ...accountDetail,
      [name]: value
    });
  };

  const resetForm = () => {
    setAccountDetail({
      name: '',
      type: accountTypes[0].value,
      balance: '',
    })
  }

  return (
    <Form className="account-form" onSubmit={handleSubmit}>
      <Form.Group>
        <Form.Input
          width={8}
          required
          label="Name"
          name="name"
          placeholder="Account name"
          value={accountDetail.name}
          onChange={handleChange}
        />
        <Form.Select
          width={8}
          label="Type"
          name="type"
          value={accountDetail.type}
          options={accountTypes}
          onChange={handleChange}
        />
      </Form.Group>
      <Form.Input
        width={8}
        type="number"
        placeholder="Balance"
        label='Balance'
        name="balance"
        value={accountDetail.balance}
        onChange={handleChange}
      />
      
      <Form.Button className="createAccountBtn" width={8} primary content="Save Account" />
    </Form>
  );
}

export default AccountForm;