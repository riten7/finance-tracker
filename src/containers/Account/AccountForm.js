import React from 'react';
import { Form } from 'semantic-ui-react';
import { accountTypes, getRandomId } from '../../utility/constant';

const AccountForm = (props) => {

  const defaultState = {
      name: '',
      type: accountTypes[0].value,
      balance: 0,
    }

  const handleSubmit = () => {
    props.handleFormSubmit({
      ...accountDetail, 
      id: getRandomId(),
    });
    resetForm();
  }

  const [accountDetail, setAccountDetail] = React.useState(defaultState);
  const handleChange = (_, { name, value }) => {
    setAccountDetail({
      ...accountDetail,
      [name]: value
    });
  };

  const resetForm = () => {
    setAccountDetail(defaultState);
  }

  return (
    <Form className="account-form" onSubmit={handleSubmit}>
      <Form.Group>
        <Form.Input
          width={8}
          required
          label="Name"
          name="name"
          placeholder="Account name"
          value={accountDetail.name}
          onChange={handleChange}
        />
        <Form.Select
          width={8}
          label="Type"
          name="type"
          value={accountDetail.type}
          options={accountTypes}
          onChange={handleChange}
        />
      </Form.Group>
      <Form.Input
        width={8}
        type="number"
        placeholder="Balance"
        label='Balance'
        name="balance"
        value={accountDetail.balance > 0 ? accountDetail.balance : '' }
        onChange={handleChange}
      />
      
      <Form.Button width={8} primary content="Save Account" />
    </Form>
  );
}

export default AccountForm;