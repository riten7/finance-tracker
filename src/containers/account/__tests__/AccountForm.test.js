import React from 'react';
import { mount } from "enzyme";
import { Provider } from 'react-redux';
import configureStore from "redux-mock-store";
import AccountForm from '../AccountForm';
import data from '../../../__mockData__/data.json';

describe("AccountForm component", () => {
  let wrapper;
  const store = configureStore()(data);
  const handleFormSubmit = jest.fn();
  it("Component renders properly", () => {
    wrapper = mount(<Provider store={store}><AccountForm handleFormSubmit={handleFormSubmit} /></Provider>);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Form submission works properly", () => {
    wrapper.find('.form').simulate('submit');
    expect(handleFormSubmit).toHaveBeenCalled();
  });
  it('Input handle change works properly', () => {
    wrapper.find('.form .input').at(0).find('input').simulate('change', { target: { value: 5000 }});
    expect(wrapper.find('.form .input').at(0).find('input').props().value).toEqual(5000);
  });
});