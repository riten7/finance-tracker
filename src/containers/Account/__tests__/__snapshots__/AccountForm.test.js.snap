// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountForm component Component renders properly 1`] = `"<form class=\\"ui form account-form\\"><div class=\\"fields\\"><div class=\\"required eight wide field\\"><label>Name</label><div class=\\"ui input\\"><input name=\\"name\\" placeholder=\\"Account name\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div></div><div class=\\"eight wide field\\"><label>Type</label><div name=\\"type\\" role=\\"listbox\\" aria-expanded=\\"false\\" class=\\"ui selection dropdown\\" tabindex=\\"0\\"><div aria-atomic=\\"true\\" aria-live=\\"polite\\" role=\\"alert\\" class=\\"divider text\\">Cash</div><i aria-hidden=\\"true\\" class=\\"dropdown icon\\"></i><div class=\\"menu transition\\"><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"true\\" aria-selected=\\"true\\" class=\\"active selected item\\"><span class=\\"text\\">Cash</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Bank Account</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Credit</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Deposit</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Asset</span></div></div></div></div></div><div class=\\"eight wide field\\"><label>Balance</label><div class=\\"ui input\\"><input placeholder=\\"Balance\\" name=\\"balance\\" type=\\"number\\" value=\\"\\"></div></div><div class=\\"field\\"><button class=\\"ui primary button\\">Save Account</button></div></form>"`;
